## Precision

To derive the correlation matrices I am assuming a discrete time series of random values. I expect that the derivation in continuous time will proceed analogously.

Assume $x$ is a stationary stochastic process. Then we can define:

\begin{align*}
\mu &= \frac{1}{n} \sum_{i=1}^{n}{x_i} \\
\sigma^2 &= 1/\pi = \frac{1}{n} \sum_{i=1}^{n}{(x_i - \bar{x})(x_i -\bar{x})} \\
\nu(h) &= \frac{1}{n} \sum_{i=1}^{n-h}{(x_i - \bar{x})(x_{i+h} - \bar{x})} \\
\rho(h) &= \nu(h)/\sigma^2 = \pi \nu(h) \\
\end{align*}

where $\mu$ is the mean, $\sigma^2$ is the variance, $\pi$ is the precision, $\nu(h)$ is the cross-covariance, and $\rho(h)$ is the autocorrelation function, both evaluated at delay $h$, with $\rho(0) = 1$ by definition, when the number of samples $n \to \infty$.

Without loss of generality, we can assume $x$ is a zero mean process so we can simplify to:

\begin{align*}
\mu &= 0 \\
\sigma^2 &= 1/\pi = \frac{1}{n} \sum_{i=1}^{n}{x_i x_i} \\
\nu(h) &= \frac{1}{n} \sum_{i=1}^{n-h}{x_i x_{i+h}} \\
\rho(h) &= \nu(h)/\sigma^2 = \pi \nu(h) \\
\end{align*}

If $x$ is not a single variable, but a set of variables, i.e., a vector $\mathbf{x}$ then the variance $\mathbf{\Sigma}$ and precision $\mathbf{\Pi}$ of the process are matrices, as are the cross-covariance $\mathbf{N}(h)$ and the autocorrelation function $\mathbf{P}(h)$ evaluated at delay $h$:

\begin{align*}
\mathbf{\Sigma}^2 &= \mathbf{\Pi}^{-1} = \frac{1}{n} \sum_{i=1}^{n-h}{\mathbf{x}_i \mathbf{x}_i^T} \\
\mathbf{N}(h) &= \frac{1}{n} \sum_{i=1}^{n-h}{\mathbf{x}_i \mathbf{x}_{i+h}} \\
\mathbf{P}(h) &= \mathbf{\Pi} \mathbf{N}(h) \\
\end{align*}

If the variables that make up $\mathbf{x}$ are independent, then $\mathbf{\Sigma}^2$ and $\mathbf{\Pi}$ are diagonal matrices. 

In **"DEM: A variational treatment of dynamic systems"**, Friston et al introduce the use of generalised coordinates containing the higher orders of motion to describe the trajectory of a time varying system in a way that allows Expectation Maximisation for dynamical systems. If $\mathbf{\tilde x}$ is a vector of generalised coordinates for state $x$, we have:

\begin{align*}
\mathbf{\tilde x} &= [x, \dot x, \ddot x, \dddot x, \cdots]^T \\
\end{align*}

where the higher orders of motion are defined as:

\begin{align*}
&\lim_{\Delta t \to 0} E \left( \dot{x}(t) - \frac{x(t + \Delta t) - x(t)}{\Delta t}\right) = 0 \\
&\lim_{\Delta t \to 0} E \left( \ddot{x}(t) - \frac{\dot{x}(t + \Delta t) - \dot{x}(t)}{\Delta t}\right) = 0 \\
&\cdots \\
\end{align*}

or, more informally, using the finite difference method:

\begin{align*}
\dot{x}(t) &= \frac{x(t + \Delta t) - x(t)}{\Delta t} ; \Delta t \to 0 \\
\ddot{x}(t) &= \frac{\dot{x}(t + \Delta t) - \dot{x}(t)}{\Delta t} \\
&= \frac{x(t + \Delta t) - x(t)}{(\Delta t)^2} - \frac{x(t) - x(t - \Delta t)}{(\Delta t)^2} \\
&= \frac{x(t + \Delta t) - 2x(t) + x(t - \Delta t)}{(\Delta t)^2}; \Delta t \to 0 \\
&\cdots\\
\end{align*}

Because the levels in generalised coordinates are derivatives of the levels above, temporal correlations will exist between the levels, hence $\mathbf{\Sigma}^2$ is no longer a diagonal matrix. We can define:

\begin{align*}
\mathbf{\Sigma}^2 = \mathbf{\Pi}^{-1} &= \frac{1}{n} \sum_{i=1}^n{\mathbf{\tilde x}_i \mathbf{\tilde x}_i^T} \\
&= 
    \begin{bmatrix}
    C(x,x) & C(x,\dot{x}) & C(x,\ddot{x}) & \cdots \\
    C(\dot{x},x) & C(\dot{x},\dot{x}) & C(\dot{x},\ddot{x}) \\
    C(\ddot{x},x) & C(\ddot{x},\dot{x}) & C(\ddot{x},\ddot{x}) \\
    \vdots & & & \ddots
    \end{bmatrix}
\end{align*}

where

\begin{align*}
C(a,b) = \frac{1}{n} \sum_{i=1}^{n}{a_i b_i}
\end{align*}



We can easily calculate the first four entries:

\begin{align*}
C(x,x) &= \sigma^2 = \nu(0) = \sigma^2 \rho(0)\\
\\
C(x,\dot{x}) = C(\dot{x},x) &= \frac{1}{n} \sum_{i=1}^{n}{x(i) \dot{x}(i)} \\
&= \frac{1}{n} \sum_{i=1}^{n}{x(i) \frac{x(i + \Delta t) - x(i)}{\Delta t}} \\
&= \frac{1}{n} \sum_{i=1}^{n}{\frac{x(i) x(i + \Delta t) - x(i) x(i)}{\Delta t}} \\
&= \frac{\nu(\Delta t) - \nu(0)}{\Delta t} \\
&= \dot{\nu}(0) \\
&= \sigma^2\dot{\rho}(0) \\
\\
C(\dot{x},\dot{x}) &= \frac{1}{n} \sum_{i=1}^{n}{\dot{x}(i) \dot{x}(i)} \\
&= \frac{1}{n} \sum_{i=1}^{n}{\frac{x(i + \Delta t) - x(i)}{\Delta t} \frac{x(i + \Delta t) - x(i)}{\Delta t}} \\
&= \frac{1}{n} \sum_{i=1}^{n}{\frac{x(i + \Delta t) x(i + \Delta t) - 2x(i) x(i + \Delta t) + x(i)x(i)}{(\Delta t)^2}} \\
&\text{and since} \; x(i + \Delta t) x(i + \Delta t) = x(i) x(i)\\
&= \frac{1}{n} \sum_{i=1}^{n}{\frac{2 x(i) x(i) - 2x(i) x(i + \Delta t)}{(\Delta t)^2}} \\
&\text{by symmetry} \; x(i + \Delta t) x(i) = x(i) x(i - \Delta t) = x(i - \Delta t) x(i)\\
&= \frac{1}{n} \sum_{i=1}^{n}{\frac{- x(i) x(i + \Delta t) + 2 x(i) x(i) - x(i) x(i - \Delta t)}{(\Delta t)^2}} \\
&= -\ddot{\nu}(0)\\
&= -\sigma^2 \ddot{\rho}(0)\\
\end{align*}

We now define:

\begin{align*}
C(\overset{j}{x},\overset{k}{x}) &= \frac{1}{n} \sum_{i=1}^{n}{\overset{j}{x}(i) \overset{k}{x}(i)} \\
&= \overset{j+k}{\nu}(0)\\
&= \sigma^2 \overset{j+k}{\rho}(0)\\
\end{align*}

where we use $\overset{j}{x}_i$ and $\overset{k}{x}_i$ to indicate the $j$th and $k$th derivative of $x_i$, respectively. We can then proceed by induction:

\begin{align*}
C(\overset{j+2}{x},\overset{k}{x}) &= \frac{1}{n} \sum_{i=1}^{n}{\overset{j+2}{x}(i) \overset{k}{x}(i)} \\
&= \frac{1}{n} \sum_{i=1}^{n}{\frac{\overset{j}{x}(i + \Delta t) - 2\overset{j}{x}(i) + \overset{j}{x}(i - \Delta t)}{(\Delta t)^2} \overset{k}{x}(i)} \\
&= \frac{1}{n} \sum_{i=1}^{n}{\frac{\overset{j}{x}(i + \Delta t) \overset{k}{x}(i) - 2\overset{j}{x}(i) \overset{k}{x}(i) + \overset{j}{x}(i - \Delta t)\overset{k}{x}(i)}{(\Delta t)^2}} \\
&= \overset{j+k+2}{\nu}(0)\\
&= \sigma^2 \overset{j+k+2}{\rho}(0)\\
\end{align*}

and symmetrically:
\begin{align*}
C(\overset{j}{x},\overset{k+2}{x}) &= \frac{1}{n} \sum_{i=1}^{n}{\overset{j}{x}(i) \overset{k+2}{x}(i)} \\
&= \frac{1}{n} \sum_{i=1}^{n}{\overset{j}{x}(i) \frac{\overset{k}{x}(i + \Delta t) - 2\overset{k}{x}(i) + \overset{k}{x}(i - \Delta t)}{(\Delta t)^2}} \\
&= \frac{1}{n} \sum_{i=1}^{n}{\frac{\overset{j}{x}(i) \overset{k}{x}(i + \Delta t) - 2\overset{j}{x}(i) \overset{k}{x}(i) + \overset{j}{x}(i)\overset{k}{x}(i - \Delta t)}{(\Delta t)^2}} \\
&= \overset{j+k+2}{\nu}(0)\\
&= \sigma^2 \overset{j+k+2}{\rho}(0)\\
\end{align*}

Thus, the entry at row $j+2$, column $k$ (both starting from $0$) is simply the second derivative of the entry at row $j$, column $k$, as is the entry at row $j$, column $k+2$. This results in the pattern that the correlation between derivatives of order $j$ and $k$ yields the $j+k$'s derivative of the autocorrelation function multiplied by $\sigma^2$ and additionally multiplied by $-1$ if both $j$ and $k$ are odd. Thus we can write:

\begin{align*}
\mathbf{\Sigma}^2 = \mathbf{\Pi}^{-1} =
    \sigma^2 \begin{bmatrix}
    \rho(0) & \dot{\rho}(0) & \ddot{\rho}(0) & \dddot{\rho}(0) & \ddot{\ddot{\rho}}(0) & \ddot{\dddot{\rho}}(0) & \cdots \\
    \dot{\rho}(0) & -\ddot{\rho}(0) & \dddot{\rho}(0) & -\ddot{\ddot{\rho}}(0) & \ddot{\dddot{\rho}}(0) & -\dddot{\dddot{\rho}}(0)\\
    \ddot{\rho}(0) & \dddot{\rho}(0) & \ddot{\ddot{\rho}}(0) & \ddot{\dddot{\rho}}(0) & \dddot{\dddot{\rho}}(0) & \dot{\dddot{\dddot{\rho}}}(0) \\
    \dddot{\rho}(0) & -\ddot{\ddot{\rho}}(0) & \ddot{\dddot{\rho}}(0) & -\dddot{\dddot{\rho}}(0) & \dot{\dddot{\dddot{\rho}}}(0) & -\ddot{\dddot{\dddot{\rho}}}(0) \\
    \ddot{\ddot{\rho}}(0) & \ddot{\dddot{\rho}}(0) & \dddot{\dddot{\rho}}(0) & \dot{\dddot{\dddot{\rho}}}(0) & \ddot{\dddot{\dddot{\rho}}}(0) & \dddot{\dddot{\dddot{\rho}}}(0) \\
    \ddot{\dddot{\rho}}(0) & -\dddot{\dddot{\rho}}(0) & \dot{\dddot{\dddot{\rho}}}(0) & -\ddot{\dddot{\dddot{\rho}}}(0) & \dddot{\dddot{\dddot{\rho}}}(0) & -\dot{\dddot{\dddot{\dddot{\rho}}}}(0)\\
    \vdots & & & & & & \ddots
    \end{bmatrix}
\end{align*}


The autocorrelation function $\rho(h)$ is by construction symmetric around $h=0$. In other words, $\rho$ is an even function. This implies directly that $\dot{\rho}(h)$ is odd, $\ddot{\rho}(h)$ is even, etc. Odd functions evaluated at $0$ yield $0$. Also $\rho(0) = 1$ as can be seen from $C(x,x)$ above. Thus we can simplify this as:

\begin{align*}
\mathbf{\Sigma}^2 = \mathbf{\Pi}^{-1} =
    \sigma^2 \begin{bmatrix}
    1 & 0 & \ddot{\rho}(0) & 0 & \ddot{\ddot{\rho}}(0) & 0 & \cdots \\
    0 & -\ddot{\rho}(0) & 0 & -\ddot{\ddot{\rho}}(0) & 0 & -\dddot{\dddot{\rho}}(0)\\
    \ddot{\rho}(0) & 0 & \ddot{\ddot{\rho}}(0) & 0 & \dddot{\dddot{\rho}}(0) & 0 \\
    0 & -\ddot{\ddot{\rho}}(0) & 0 & -\dddot{\dddot{\rho}}(0) & 0 & -\ddot{\dddot{\dddot{\rho}}}(0) \\
    \ddot{\ddot{\rho}}(0) & 0 & \dddot{\dddot{\rho}}(0) & 0 & \ddot{\dddot{\dddot{\rho}}}(0) & 0 \\
    0 & -\dddot{\dddot{\rho}}(0) & 0 & -\ddot{\dddot{\dddot{\rho}}}(0) & 0 & -\dot{\dddot{\dddot{\dddot{\rho}}}}(0)\\
    \vdots & & & & & & \ddots
    \end{bmatrix}
\end{align*}

This matrix can be evaluated for any analytic autocorrelation function. If we assume, for convenience, that the temporal correlation of all innovations have the same zero mean Gaussian form with precision parameter $\gamma = 1/\sigma^2$, we can proceed with:

\begin{align*}
\rho(h) &= e^{-\frac{\gamma}{2} h^2} \\
\dot{\rho}(h) &= -\gamma h \rho(h) \\
\ddot{\rho}(h) &= -\gamma \rho(h) + (\gamma h)^2 \rho(h) \\
\dddot{\rho}(h) &= \gamma^2 h \rho(h) + 2 \gamma^2 h \rho(h) - (\gamma h)^3 \rho(h) \\
&= 3 \gamma^2 h \rho(h) - (\gamma h)^3 \rho(h) \\
\ddot{\ddot{\rho}}(h) &= 3 \gamma^2 \rho(h) - 3 \gamma^3 h^2 \rho(h) - 3 \gamma^3 h^2 \rho(h) + (\gamma h)^4 \rho(h) \\
&= 3 \gamma^2 \rho(h) - 6 \gamma^3 h^2 \rho(h) + (\gamma h)^4 \rho(h) \\
\ddot{\dddot{\rho}}(h) &= - 3 \gamma^3 h \rho(h) - 12 \gamma^3 h \rho(h) + 6 \gamma^4 h^3 \rho(h) + 4 \gamma^4 h^3 \rho(h) - (\gamma h)^5 \rho(h) \\
&= - 15 \gamma^3 h \rho(h) + 10 \gamma^4 h^3 \rho(h) - (\gamma h)^5 \rho(h) \\
\dddot{\dddot{\rho}}(h) &= -15 \gamma^3 \rho(h) + 15 \gamma^4 h^2 \rho(h) + 30 \gamma^4 h^2 \rho(h) - 10 \gamma^5 h^4 \rho(h) - 5 \gamma^5 h^4 \rho(h) + (\gamma h)^6 \rho(h) \\
&= -15 \gamma^3 \rho(h) + 45 \gamma^4 h^2 \rho(h) - 15 \gamma^5 h^4 \rho(h) + (\gamma h)^6 \rho(h) \\
\end{align*}

The expression for higher order derivatives gets longer and longer. As a shortcut to calculate the higher order derivatives, compare the expressions above with those of the (probabilist's) Hermite polynomials:

\begin{align*}
He_n(x) &= (-1)^n e^{\frac{x^2}{2}} \frac{d^n}{dx^n} e^{-\frac{x^2}{2}} \\
\rho(h) &= e^{-\frac{\gamma}{2} h^2} \\
&= \frac{He_0(\gamma^{1/2} h)}{(-1)^0 e^{\frac{\gamma h^2}{2}}} \\
\overset{n}\rho(h) &= \frac{He_n(\gamma^{1/2} h)}{(-1)^n e^{\frac{\gamma h^2}{2}}} \\
\end{align*}

where $\overset{n}\rho(h)$ is the $n$th derivative of $\rho(h)$. Because we are only interested in the values evaluated at $h=0$ for even values of $n$, the denominator evaluates to $1$ so that this simplifies to:

\begin{align*}
\overset{n}\rho(0) &= He_n(0) \\
\end{align*}

The explicit expression for the Hermite polynomials can be used to calculate the coefficients for all even derivatives quickly. For $n$ is even, they are given by

\begin{align*}
He_n(x) &= n!\sum_{m=0}^{n/2}\frac{(-1)^m}{m! (n-2m)!}\frac{x^{n-2m}}{2^m} \\
He_n(0) &= n!\sum_{m=0}^{n/2}\frac{(-1)^m}{m! (n-2m)!}\frac{0^{n-2m}}{2^m} \\
 &= n!\frac{(-1)^{n/2}}{(n/2)! (0)!}\frac{0^0}{2^{n/2}} \\
 &= n!\frac{(-1)^{n/2}}{(n/2)!}\frac{1}{2^{n/2}} \\
 &= \left(-\frac{1}{2}\right)^{n/2}\frac{n!}{(n/2)!}\\
\end{align*}


Evaluating the above yields:


In [2]:
from math import factorial as fac
print("Derivatives of 𝜌")
for m in range(6):
    print("{:<3} {:>8} γ^{:}".format(2*m, (-1/2)**m * fac(2*m) / fac(m), m))

Derivatives of 𝜌
0        1.0 γ^0
2       -1.0 γ^1
4        3.0 γ^2
6      -15.0 γ^3
8      105.0 γ^4
10    -945.0 γ^5


\begin{align*}
\mathbf{\Sigma}^2 = \mathbf{\Pi}^{-1} &=
    \sigma^2 \begin{bmatrix}
    1 & 0 & \ddot{\rho}(0) & 0 & \ddot{\ddot{\rho}}(0) & 0 & \cdots \\
    0 & -\ddot{\rho}(0) & 0 & -\ddot{\ddot{\rho}}(0) & 0 & -\dddot{\dddot{\rho}}(0)\\
    \ddot{\rho}(0) & 0 & \ddot{\ddot{\rho}}(0) & 0 & \dddot{\dddot{\rho}}(0) & 0 \\
    0 & -\ddot{\ddot{\rho}}(0) & 0 & -\dddot{\dddot{\rho}}(0) & 0 & -\ddot{\dddot{\dddot{\rho}}}(0) \\
    \ddot{\ddot{\rho}}(0) & 0 & \dddot{\dddot{\rho}}(0) & 0 & \ddot{\dddot{\dddot{\rho}}}(0) & 0 \\
    0 & -\dddot{\dddot{\rho}}(0) & 0 & -\ddot{\dddot{\dddot{\rho}}}(0) & 0 & -\dot{\dddot{\dddot{\dddot{\rho}}}}(0)\\
    \vdots & & & & & & \ddots
    \end{bmatrix} \\
 &= \sigma^2 \begin{bmatrix}
    1 & 0 & -\gamma & 0 & 3 \gamma^2 & 0 \\
    0 & \gamma & 0 & -3 \gamma^2 & 0 & 15 \gamma^3 \\
    -\gamma & 0 & 3 \gamma^2 & 0 & -15 \gamma^3 & 0 \\
    0 & -3 \gamma^2 & 0 & 15 \gamma^3 & 0 & -105 \gamma^4 \\
    3 \gamma^2 & 0 & -15 \gamma^3 & 0 & 105 \gamma^4 & 0 \\
    0 & 15 \gamma^3 & 0 & -105 \gamma^4 & 0 & 945 \gamma^5 \\
    \end{bmatrix} \\
\end{align*}

Note, to get Friston's version, we need to replace $\gamma$ with $\gamma / 2$. In his work he simply states that he assumes a "Gaussian form" for the autocorrelation with precision parameter $\gamma$. The autocorrelation functions isn't explicitly defined, but in order to get the results in the paper, it would have to be defined as $\rho(h) = \exp(-\frac{\gamma}{4} h^2)$, which implies that $\gamma$ is the precision of the process multiplied by two. This seems an unusual definition, but yields the results in his papers:

\begin{align*}
\mathbf{\Sigma}^2 = \mathbf{\Pi}^{-1}
 &= \sigma^2 \begin{bmatrix}
    1 & 0 & -\frac{\gamma}{2} & 0 & 3 \left(\frac{\gamma}{2}\right)^2 & 0 \\
    0 & \frac{\gamma}{2} & 0 & -3 \left(\frac{\gamma}{2}\right)^2 & 0 & 15 \left(\frac{\gamma}{2}\right)^3 \\
    -\frac{\gamma}{2} & 0 & 3 \left(\frac{\gamma}{2}\right)^2 & 0 & -15 \left(\frac{\gamma}{2}\right)^3 & 0 \\
    0 & -3 \left(\frac{\gamma}{2}\right)^2 & 0 & 15 \left(\frac{\gamma}{2}\right)^3 & 0 & -105 \left(\frac{\gamma}{2}\right)^4 \\
    3 \left(\frac{\gamma}{2}\right)^2 & 0 & -15 \left(\frac{\gamma}{2}\right)^3 & 0 & 105 \left(\frac{\gamma}{2}\right)^4 & 0 \\
    0 & 15 \left(\frac{\gamma}{2}\right)^3 & 0 & -105 \left(\frac{\gamma}{2}\right)^4 & 0 & 945 \left(\frac{\gamma}{2}\right)^5 \\
    \end{bmatrix} \\
 &= \sigma^2 \begin{bmatrix}
    1 & 0 & -\frac{1}{2}\gamma & 0 & \frac{3}{4}\gamma^2 & 0 \\
    0 & \frac{1}{2}\gamma & 0 & -\frac{3}{4}\gamma^2 & 0 & \frac{15}{8}\gamma^3 \\
    -\frac{1}{2}\gamma & 0 & \frac{3}{4}\gamma^2 & 0 & -\frac{15}{8}\gamma^3 & 0 \\
    0 & -\frac{3}{4}\gamma^2 & 0 & \frac{15}{8}\gamma^3 & 0 & -\frac{105}{16}\gamma^4 \\
    \frac{3}{4}\gamma^2 & 0 & -\frac{15}{8}\gamma^3 & 0 & \frac{105}{16}\gamma^4 & 0 \\
    0 & \frac{15}{8}\gamma^3 & 0 & -\frac{105}{16}\gamma^4 & 0 & \frac{945}{32}\gamma^5 \\
    \end{bmatrix} \\
\end{align*}

Note, the literature on Hermite polynomials can cause confusion because the physicist's definition and probabilist's definition are different. If we define the autocorrelation as $\rho(h) = \exp(-\gamma h^2)$, we could then use the physicist's Hermite polynomials instead.

\begin{align*}
H_n(x) &= (-1)^n e^{x^2} \frac{d^n}{dx^n} e^{-x^2} \\
\rho(h) &= e^{-\gamma h^2} \\
&= \frac{H_0(\gamma^{1/2} h)}{(-1)^0 e^{\frac{\gamma h^2}{2}}} \\
\overset{n}\rho(h) &= \frac{H_n(\gamma^{1/2} h)}{(-1)^n e^{\frac{\gamma h^2}{2}}} \\
\end{align*}

Evaluating these at $h=0$ for even values of $n$, the denominator evaluates to $1$ so that this simplifies to:

\begin{align*}
\overset{n}\rho(0) &= H_n(0) \\
\end{align*}

The explicit expression for the Hermite polynomials can be used to calculate the coefficients for all even derivatives quickly. For $n$ is even, they are given by

\begin{align*}
H_n(x) &= n!\sum_{m=0}^{n/2}\frac{(-1)^{n/2 - m}}{2m! (n/2-m)!} (2x)^{2m} \\
H_n(0) &= n!\sum_{m=0}^{n/2}\frac{(-1)^{n/2 - m}}{2m! (n/2-m)!} (0)^{2m} \\
 &= n!\frac{(-1)^{n/2}}{(0)! (n/2)!} 0^0 \\
 &= (-1)^{n/2}\frac{n!}{(n/2)!} \\
\end{align*}

In [3]:
print("Derivatives of 𝜌")
for m in range(6):
    print("{:<3} {:>8} γ^{:}".format(2*m, (-1)**m * fac(2*m) / fac(m), m))

Derivatives of 𝜌
0        1.0 γ^0
2       -2.0 γ^1
4       12.0 γ^2
6     -120.0 γ^3
8     1680.0 γ^4
10  -30240.0 γ^5


\begin{align*}
\mathbf{\Sigma}^2 = \mathbf{\Pi}^{-1} &=
    \sigma^2 \begin{bmatrix}
    1 & 0 & \ddot{\rho}(0) & 0 & \ddot{\ddot{\rho}}(0) & 0 & \cdots \\
    0 & -\ddot{\rho}(0) & 0 & -\ddot{\ddot{\rho}}(0) & 0 & -\dddot{\dddot{\rho}}(0)\\
    \ddot{\rho}(0) & 0 & \ddot{\ddot{\rho}}(0) & 0 & \dddot{\dddot{\rho}}(0) & 0 \\
    0 & -\ddot{\ddot{\rho}}(0) & 0 & -\dddot{\dddot{\rho}}(0) & 0 & -\ddot{\dddot{\dddot{\rho}}}(0) \\
    \ddot{\ddot{\rho}}(0) & 0 & \dddot{\dddot{\rho}}(0) & 0 & \ddot{\dddot{\dddot{\rho}}}(0) & 0 \\
    0 & -\dddot{\dddot{\rho}}(0) & 0 & -\ddot{\dddot{\dddot{\rho}}}(0) & 0 & -\dot{\dddot{\dddot{\dddot{\rho}}}}(0)\\
    \vdots & & & & & & \ddots
    \end{bmatrix} \\
 &= \sigma^2 \begin{bmatrix}
    1 & 0 & -2\gamma & 0 & 12 \gamma^2 & 0 \\
    0 & 2\gamma & 0 & -12 \gamma^2 & 0 & 120 \gamma^3 \\
    -2\gamma & 0 & 12 \gamma^2 & 0 & -120 \gamma^3 & 0 \\
    0 & -12 \gamma^2 & 0 & 120 \gamma^3 & 0 & -1680 \gamma^4 \\
    12 \gamma^2 & 0 & -120 \gamma^3 & 0 & 1680 \gamma^4 & 0 \\
    0 & 120 \gamma^3 & 0 & -1680 \gamma^4 & 0 & 30240 \gamma^5 \\
    \end{bmatrix} \\
\end{align*}